### PR TITLE
Add new resource for Jenkins credential Azure Service Principal

### PR DESCRIPTION
**Pull request by gitflo1**
# Dependencies

The new resource requires the Jenkins Plugin [Azure Credentials Plugin](https://plugins.jenkins.io/azure-credentials/).

# What Is Changing

The PR is addressing the issue #107.  A new resource, named `jenkins_credential_azure_service_principal` will be created.

# Test Steps

- [x] Have you updated the `docs/` folder to match your change?
- [x] Have you exercised your change using the `examples/` docker compose setup?

```sh
make testacc
```

### CODE REVIEW

**Comment by gitflo1** [comment]
@taiidani could you please approve the workflow for acceptance tests?

**Comment by gitflo1** [comment]
Unfortunately, I faced some minor character interpretation issues on my local WindowsOS. Nevertheless, I was able to perform a `terraform apply` with my local setup. I hope this is sufficient for now:
image

**Comment by gitflo1** [comment]
Any updates @taiidani ?

**Comment by taiidani** [comment]
@gitflo1 Hey there, sorry for the delay!

I've performed a code review and really like the work being put in here. Overall the changes LGTM and check all the boxes that I look for in contributions. However, I ran into some trouble trying to test the new resource.

Locally I've been receiving the following error when applying the new resource in `example/`:
```
Error: Could not create Azure service principal credentials: invalid response code 500
```

I attempted to recreate this by adding a new `example` CI test in `main` and pushing [a PR](https://github.com/taiidani/terraform-provider-jenkins/pull/114) including your changes, which is also failing but with a different error.

Is it possible that the Jenkins Plugin is attempting to reach out to Azure during the `apply`? If confirmed so, we could simulate Azure locally or, given these are service credentials and not expensive, use a dummy Azure account for the integration test?

**Comment by gitflo1** [comment]
@taiidani Thank you for the review and your testing efforts!
Afaik the Azure Plugin is not reaching out to Azure in order to check the existence of the Azure Service Principal created. In the Azure plugin's UI exists an additional button called "verify service principal" that does exactly this job (see image attached). 
image

Are you sure that the Azure Plugin is actually installed in your local Docker Jenkins test instance? Can you send me the Terraform file producing the error when applied?

File: example/credentials_azure_service_principal.tf [hunk 1]
```diff
@@ -0,0 +1,14 @@
+resource "jenkins_folder" "azure_service_principal_test_folder" {
+    name        = "azure-service-principal-test-folder"
+    description = "A sample folder"
+}
```
Comment (taiidani) [inline example/credentials_azure_service_principal.tf:4]: It doesn't appear as though this folder is being used. Is it needed?

**Review by taiidani — CHANGES_REQUESTED** [review]
@gitflo1 Interesting! After poking at it a few more times, I managed to get it to work. It may be a race condition in the job that runs the [example](https://github.com/taiidani/terraform-provider-jenkins/tree/main/example) TF. I had not added any new TF beyond those already committed to your PR.

One very small ask that I noticed during my second review, but otherwise I think I can chalk this up to the `example` CI automation being flakey. I'll take responsibility for ensuring that continues to pass post-merge.

Thanks again!

File: example/credentials_azure_service_principal.tf (hunk 1 above)
Comment (gitflo1) [reply example/credentials_azure_service_principal.tf:4]: Thank you for the hint. I intentionally created the new folder as test folder for the `jenkins_credential_azure_service_principal` test.  I just updated the usage here. But if you want, I can use the example folder here. Can we resolve the conversation or should I update to the example folder and remove the new test folder?

Comment (taiidani) [reply example/credentials_azure_service_principal.tf:4]: I think I like it going under the example folder -- that way it's a little more clear where all of the example credentials are being Terraformed into. Would you mind updating to use that and remove the `azure-service-principal-test-folder` folder?

**Comment by taiidani** [comment]
The new `example` test failed, but per the discussion above I'll be taking responsibility for ensuring that it succeeds post-merge. I have started https://github.com/taiidani/terraform-provider-jenkins/pull/115 to begin that work.

File: example/credentials_azure_service_principal.tf (hunk 1 above)
Comment (gitflo1) [reply example/credentials_azure_service_principal.tf]: Sure. Updated it to use the example folder and removed the other folder.

**Review by taiidani — APPROVED** [review]
LGTM! Thanks for your contribution and working with me on the updates!

**Pull request merged**